src/setup-beam.js

```diff
@@ -413,38 +413,37 @@ function getRunnerOSVersion() {
 }
 
 async function get(url0, pageIdxs) {
-  function getPage(pageIdx) {
+  function getPage(pageIdx, retries0) {
+    let retries = retries0
     return new Promise((resolve, reject) => {
+      if (retries === undefined) {
+        retries = 5
+      }
       const url = new URL(url0)
       if (pageIdx !== null) {
         url.searchParams.append('page', pageIdx)
       }
-      https
-        .get(
-          url,
-          {
-            headers: { 'user-agent': 'setup-beam' },
-          },
-          (res) => {
-            let data = ''
-            res.on('data', (chunk) => {
-              data += chunk
-            })
-            res.on('end', () => {
-              if (res.statusCode >= 400 && res.statusCode <= 599) {
-                reject(
-                  new Error(
-                    `Got ${res.statusCode} from ${url}. Exiting with error`,
-                  ),
-                )
-              } else {
-                resolve(data)
-              }
-            })
-          },
-        )
-        .on('error', (err) => {
-          reject(err)
+      request({
+        url: url.toString(),
+        headers: {
+          authorization: `Bearer ${process.env.GITHUB_TOKEN}`,
```

Comment:
⚠️ 👀 ⚠️ 👀 ⚠️ 👀 ⚠️ 👀 

I'm not sure if this (accompanied by the changes in the GitHub workflows) presents any type of security issue, which is why I ask for many eyes on this.

Reply:
In principle it should be fine, but I don't feel great about it either 🤔

Reply:
A fresh GITHUB_TOKEN is created on each workflow run, and expired after it is completed: https://docs.github.com/en/actions/security-guides/automatic-token-authentication

The token does not have permission (by default) to modify the repository, but mostly CI things:
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Also note from the GitHub docs:

> Important: An action can access the GITHUB_TOKEN through the github.token context even if the workflow does not explicitly pass the GITHUB_TOKEN to the action. As a good security practice, you should always make sure that actions only have the minimum access they require by limiting the permissions granted to the GITHUB_TOKEN.

So the action already theoretically has access to the GITHUB_TOKEN even before explicitly reading it from the process environment. So users depending on actions (and writing their own) should be limiting the scope via CI config anyway, which can be done on at least a per job basis.